### PR TITLE
Qualification tool: Read SQL function names for parsing expressions

### PR DESCRIPTION
**Pull request by nartal1**
This fixes a bug where some of the expressions were marked as not supported even though those were supported in spark-rapids.
Earlier we were reading Expression name and saving that as key which was later used to check if a particular expression is supported or not.
Issue was for some of the expressions, their corresponding SQL function names were different. In the eventlog, SQL function name is present which is parsed in the tool.
Examples: 
1. Expression name - Average, SQL function name - avg
2. Expression name -Substring, SQL function name - substr; substring

This PR fixes the bug where it reads the SQL function names as well and stores them as key in addition to Expression names.
Added unit test which verfies reading of SQL function name and parsing it correctly.

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/PluginTypeChecker.scala [hunk 1]
```diff
@@ -94,7 +94,10 @@ class PluginTypeChecker extends Logging {
 
   private def readSupportedExprs: Map[String, String] = {
     val source = Source.fromResource(SUPPORTED_EXPRS_FILE)
-    readSupportedOperators(source).map(x => (x._1.toLowerCase, x._2))
+    // Some SQL function names have backquotes(`) around their names,
+    // so we remove them before saving.
+      readSupportedOperators(source, "exprs").map(
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/PluginTypeChecker.scala:99]: nit spacing , indented to far

**Comment by nartal1** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Comment by nartal1** [comment]
build

**Pull request merged**